rero_ils/modules/acquisition/rollover.py

```diff
@@ -0,0 +1,664 @@
+    #  use outside this class
+
+    def _abort_rollover(self, message=None):
+        """Aborting the rollover process.
+
+        This will delete all acquisition resources created on the destination
+        `Budget` resource in the reverse order of their creation.
+
+        :param message: the message to log.
+        """
+        if message:
+            self.logger.warning(message)
+        self.logger.warning('Aborting rollover process !')
+        self._update_budgets(True, False)
+        if not self.stack:
+            return
+        self.logger.info('Purging created resources...')
+        for obj in reversed(self.stack):
+            obj.delete(force=True, dbcommit=True, delindex=True)
+            self.logger.info(f"\t* object {str(obj)} deleted")
+
+    def _confirm(self, question, default="yes"):
+        """Ask a yes/no question via raw_input() and return their answer.
+
+        :param question: is a string that is presented to the user.
+        :param default: is the presumed answer if the user just hits <Enter>.
+            It must be "yes" (the default), "no" or None (meaning an answer
+            is required of the user).
+        :return: The "answer" return value is True for "yes" or False for "no".
+        """
+        valid = {"yes": True, "y": True, "ye": True, "no": False, "n": False}
+        if default is None:
+            prompt = " [y/n] "
+        elif default == "yes":
+            prompt = " [Y/n] "
+        elif default == "no":
+            prompt = " [y/N] "
+        else:
+            raise ValueError("invalid default answer: '%s'" % default)
+
+        while True:
+            self.logger.info(question + prompt)
+            choice = input().lower()
+            if default is not None and choice == "":
+                return valid[default]
+            elif choice in valid:
+                return valid[choice]
+            else:
+                self.logger.warning("Please respond with 'yes' or 'no'.")
+
+    def _create_new_budget(self, **kwargs):
+        """Create a new budget to use for rollover setting.
+
+        :param organisation_pid: the organisation pid for the new budget.
+        :param **kwargs: all necessary information to create the new budget.
+        :return: the new `Budget` resource.
+        :raises ValueError: If some required budget parameters are missing.
+        """
+        org_pid = self.original_budget.organisation_pid
+        data = {
+            'organisation': {'$ref': get_ref_for_pid('org', org_pid)},
+            'is_active': False
+        }
+        for required_param in ['name', 'start_date', 'end_date']:
+            assert required_param in kwargs, f'{required_param} param required'
+            data[required_param] = kwargs[required_param]
+        if budget := Budget.create(data, dbcommit=True, reindex=True):
+            self.stack.append(budget)
+        return budget
+
+    def _draw_data_table(self, columns, rows=[], padding=''):
```

Comment:
Did you consider to use an existing python package such as https://pypi.org/project/beautifultable or make a separate class or function for other purpose.

Reply:
I tried to use this library. 
* Table is less useful to read (due to extra line separator)
* 1 more package with 2 dependencies
* More complex (but more flexible) to use/write

I keep my 'simple' function but place it into the general `utils.py` file to be reused if needed.